schemahcl/hcl.go

```diff
@@ -541,6 +538,49 @@ func findAttr(attrs []*Attr, k string) (*Attr, bool) {
 	return nil, false
 }
 
+func hclRefTokens(ref string) hclwrite.Tokens {
+	s := strings.Split(strings.ReplaceAll(ref, "$", ""), ".")
+	t := make([]*hclwrite.Token, 0, len(s))
+	for i, s := range s {
+		if i == 0 {
+			t = append(t, hclRawTokens(s)...)
+		} else if hclsyntax.ValidIdentifier(s) {
+			t = append(t, hclTraverseAttr(s)...)
+		} else {
+			t = append(t, hclTraverseIndex(cty.StringVal(s))...)
+		}
+	}
+	return t
+}
+
+func hclTraverseAttr(name string) hclwrite.Tokens {
```

Comment:
Why "traverse"?

Reply:
I used the term from `NewExpressionAbsTraversal`  https://github.com/hashicorp/hcl/blob/v2.13.0/hclwrite/ast_expression.go#L87-L119